lib/recurly/pricing/subscription/index.js

```diff
@@ -334,6 +335,31 @@ export default class SubscriptionPricing extends Pricing {
       resolve(code);
     }, this).nodeify(done);
   }
+
+  /**
+   * Ensures a plan is set before repricing
+   *
+   * @return {PricingPromise}
+   */
+  reprice (...args) {
+    return new PricingPromise((resolve, reject) => {
+      if (!this.items.plan) return this.error(errors('missing-plan'), reject, 'plan');
+      else return resolve(super.reprice(...args));
+    });
+  }
+
+  /**
+   * Checks whether a given coupon may apply to this subscription
+   * @param  {Object} coupon
+   * @return {Boolean}
+   */
+  couponIsValidForSubscription (coupon) {
+    if (!coupon) return false;
+    if (!coupon.applies_to_plans) return false;
+    if (coupon.applies_to_all_plans) return true;
```

Comment:
another option would be to consolidate these last three lines into:

```
return coupon.applies_to_all_plans || ~coupon.plans.indexOf(this.items.plan.code)
```

Reply:
For business rules like this, I personally prefer the readability of the early returns